sdk/cosmos/azure-cosmos-spark_3-4_2-12/src/main/scala/com/azure/cosmos/spark/CosmosRowConverter.scala

```diff
@@ -0,0 +1,134 @@
+// Licensed under the MIT License.
+package com.azure.cosmos.spark
+
+import com.azure.cosmos.spark.SchemaConversionModes.SchemaConversionMode
+import com.fasterxml.jackson.annotation.JsonInclude.Include
+// scalastyle:off underscore.import
+import com.fasterxml.jackson.databind.node._
+import com.fasterxml.jackson.databind.{JsonNode, ObjectMapper}
+import java.time.format.DateTimeFormatter
+import java.time.{LocalDateTime}
+import scala.collection.concurrent.TrieMap
+
+// scalastyle:off underscore.import
+import org.apache.spark.sql.types._
+// scalastyle:on underscore.import
+
+import scala.util.{Try, Success, Failure}
+
+// scalastyle:off
+private[cosmos] object CosmosRowConverter {
+
+    // TODO: Expose configuration to handle duplicate fields
+    // See: https://github.com/Azure/azure-sdk-for-java/pull/18642#discussion_r558638474
+    private val rowConverterMap = new TrieMap[CosmosSerializationConfig, CosmosRowConverter]
+
+    def get(serializationConfig: CosmosSerializationConfig): CosmosRowConverter = {
+        rowConverterMap.get(serializationConfig) match {
+            case Some(existingRowConverter) => existingRowConverter
+            case None =>
+                val newRowConverterCandidate = createRowConverter(serializationConfig)
+                rowConverterMap.putIfAbsent(serializationConfig, newRowConverterCandidate) match {
+                    case Some(existingConcurrentlyCreatedRowConverter) => existingConcurrentlyCreatedRowConverter
+                    case None => newRowConverterCandidate
+                }
+        }
+    }
+
+    private def createRowConverter(serializationConfig: CosmosSerializationConfig): CosmosRowConverter = {
+        val objectMapper = new ObjectMapper()
+        import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule
+        objectMapper.registerModule(new JavaTimeModule)
+        serializationConfig.serializationInclusionMode match {
+            case SerializationInclusionModes.NonNull => objectMapper.setSerializationInclusion(Include.NON_NULL)
+            case SerializationInclusionModes.NonEmpty => objectMapper.setSerializationInclusion(Include.NON_EMPTY)
+            case SerializationInclusionModes.NonDefault => objectMapper.setSerializationInclusion(Include.NON_DEFAULT)
+            case _ => objectMapper.setSerializationInclusion(Include.ALWAYS)
+        }
+
+        new CosmosRowConverter(objectMapper, serializationConfig)
+    }
+}
+
+private[cosmos] class CosmosRowConverter(private val objectMapper: ObjectMapper, private val serializationConfig: CosmosSerializationConfig)
+    extends CosmosRowConverterBase(objectMapper, serializationConfig) {
+
+    override def convertSparkDataTypeToJsonNodeConditionallyForSparkRuntimeSpecificDataType
+    (
+        fieldType: DataType,
+        rowData: Any
+    ): Option[JsonNode] = {
+        fieldType match {
+            case TimestampNTZType => convertToJsonNodeConditionally(rowData.asInstanceOf[java.time.LocalDateTime].toString)
```

Comment:
i Think this should be case TimestampNTZType if rowData.isInstanceOf[java.time.ocalDateTime] - to make sure if rowData is of a different type we get the catch all case exception instead of some invalid cast exception.

Reply:
Ok, will add